subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -48,66 +40,106 @@ static struct bt_mesh_silvair_enocean_srv *model_resolve(const bt_addr_le_t *add
 	return NULL;
 }
 
-static uint32_t delay_get(struct bt_mesh_model_pub *pub, uint32_t pub_num)
-{
-	return (BT_MESH_PUB_MSG_TOTAL(pub) - pub_num) * 50;
-}
-
-static void send_onoff(struct bt_mesh_silvair_enocean_button *button,
-		       bool on_off)
+static void send_delta(struct bt_mesh_silvair_enocean_button *button, uint32_t delay,
+		       bool new_transaction)
 {
 	struct bt_mesh_model_transition transition;
-	struct bt_mesh_onoff_set msg = {
-		.on_off = on_off,
+	struct bt_mesh_lvl_delta_set msg = {
 		.transition = &transition,
+		.new_transaction = new_transaction,
+		.delta = button->delta
+	};
+	struct bt_mesh_msg_ctx ctx = {
+		.addr = button->onoff.pub.addr,
+		.send_ttl = button->onoff.pub.ttl,
+		.app_idx = button->onoff.pub.key
 	};
 
-	button->onoff.pub.retransmit = BT_MESH_PUB_TRANSMIT(3, 50);
+	if (CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TRANSITION_TIME == -1) {
+		bt_mesh_dtt_srv_transition_get(button->lvl.model, &transition);
+	} else {
+		transition.time = CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA_TRANSITION_TIME;
+	}
 
-	bt_mesh_dtt_srv_transition_get(button->onoff.model, &transition);
-	transition.delay = delay_get(&button->onoff.pub, 1);
+	transition.delay = delay;
 
-	bt_mesh_onoff_cli_set_unack(&button->onoff, NULL, &msg);
+	bt_mesh_lvl_cli_delta_set_unack(&button->lvl, &ctx, &msg);
 }
 
-static void send_delta(struct bt_mesh_silvair_enocean_button *button, enum phase phase)
+static void phase_b_tick(struct bt_mesh_silvair_enocean_button *button)
 {
-	struct bt_mesh_model_transition transition;
-	int32_t delta = CONFIG_BT_MESH_SILVAIR_ENOCEAN_DELTA *
-		(1 + button->tick_count);
-
-	if (!button->target) {
-		delta *= -1;
+	send_delta(button, 150 - (50 * button->tick_count), button->tick_count == 0);
+	if (button->tick_count++ < 3) {
+		k_work_reschedule(&button->timer, K_MSEC(50));
+	} else if (button->release_pending) {
+		/** Send only 4 messages in Phase D, because the last message sent in the
+		 *  Phase B tick handler will be counted as the first of the five in phase D
+		 *  (see log sample in spec).
+		 */
+		button->tick_count = 1;
+		button->state = BT_MESH_SILVAIR_ENOCEAN_STATE_PHASE_D;
+		k_work_reschedule(&button->timer, K_MSEC(50));
+	} else {
+		button->state = BT_MESH_SILVAIR_ENOCEAN_STATE_PHASE_C;
+		button->tick_count = 0;
+		k_work_reschedule(&button->timer,
+				  K_MSEC(CONFIG_BT_MESH_SILVAIR_ENOCEAN_TICK_TIME));
```

Comment:
I guess this is not a part of this PR directly, but is there any reason why `CONFIG_BT_MESH_SILVAIR_ENOCEAN_TICK_TIME` is a configurable parameter? As far as I can see from the spec, there is no mention of using any other value than the default value in phase C, e.g. altering it breaks the spec.

Reply:
Like I mentioned in my response to Omkar, I don't want to remove any kconfig options in this commit as it might go in a patch release. I agree with your point, I'll make a second PR removing all the unecessary options after this.